src/frontend/src/optimizer/rule/pull_up_correlated_predicate_agg_rule.rs

```diff
@@ -153,55 +153,32 @@ impl Rule for PullUpCorrelatedPredicateAggRule {
 
         let new_bottom_proj: PlanRef = LogicalProject::new(filter, bottom_proj_exprs).into();
 
-        // If there is a count aggregate, bail out and leave for general subquery unnesting to deal.
-        // When group by is empty, count would return 0 instead of null.
-        // Unless we can prove that the row generated by the empty group could be eliminated by the top filter.
-        //
-        // A typical example is the count is generate by avg, because avg = sum / count. In this case, when input is empty,
-        // sum is null, so avg is null. And null-rejected expression will be false, so we can still apply this rule and we don't need to generate a 0 value for count.
-        let count_exists = agg_calls
-            .iter()
-            .any(|agg_call| matches!(agg_call.agg_type, AggType::Builtin(PbAggKind::Count)));
-
-        if count_exists {
-            // When group input is empty, not count agg would return null.
+        // If the `group by + proj` doesn't return null for empty input, we can't apply this rule.
+        {
+            // When group input is empty, if the agg is not `count`, it would return null.
             let null_agg_pos = agg_calls
                 .iter()
                 .positions(|agg_call| {
                     !matches!(agg_call.agg_type, AggType::Builtin(PbAggKind::Count))
                 })
                 .collect_vec();
 
```

Comment:
Shall we add `array` aggregation here?

Reply:
array subquery is planned as `array_agg + coalesce`

Reply:
and `array_agg` is null-propagating

Reply:
OK, I see.